features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/event/RumEventMeta.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain.event
+
+import com.google.gson.JsonObject
+import java.util.Locale
+import kotlin.jvm.Throws
+
+internal sealed class RumEventMeta {
+
+    abstract val type: String
+
+    open fun toJson(): JsonObject {
+        val model = JsonObject()
+
+        model.addProperty(TYPE_KEY, type)
+
+        return model
+    }
+
+    companion object {
+
+        private const val UNKNOWN_RUM_EVENT_META_TYPE_ERROR = "Unknown RUM event meta type value [%s]"
+
+        const val TYPE_KEY = "type"
+        const val VIEW_TYPE_VALUE = "view"
+        const val VIEW_ID_KEY = "viewId"
+        const val DOCUMENT_VERSION_KEY = "documentVersion"
+
+        @Throws(
+            NullPointerException::class,
+            ClassCastException::class,
+            IllegalStateException::class,
+            NumberFormatException::class
+        )
+        fun fromJson(model: JsonObject): RumEventMeta {
+            return when (val type = model.get(TYPE_KEY).asString) {
+                VIEW_TYPE_VALUE -> {
+                    val viewId = model.get(VIEW_ID_KEY).asString
+                    val docVersion = model.get(DOCUMENT_VERSION_KEY).asLong
+
+                    View(viewId, docVersion)
```

Comment:
> [!NOTE]
> It might be more readable (especially when we have more metadata types) to have a `RumEventMeta.View.fromJson(JsonObject)` companion method instead? That way the `RumEventMeta` companion doesn't need to know about the `View` structure?

Reply:
I would rather keep it in the companion of `RumEventMeta`, because if we add more meta types in the future we then leak responsibility - external caller should decide companion method of which class exactly to call, and to know that it needs to parse meta into JSON to know exact metadata type, so it means we are doing JSON parsing for the same meta binary in 2 places.